test/functional/wallet_resendwallettransactions.py

```diff
@@ -49,16 +49,21 @@ def run_test(self):
         block.solve()
         node.submitblock(ToHex(block))
 
-        # Transaction should not be rebroadcast
         node.syncwithvalidationinterfacequeue()
-        node.p2ps[1].sync_with_ping()
+        now = int(time.time())
+
+        # Transaction should not be rebroadcast within first 12 hours
+        # Leave 2 mins for buffer
+        twelve_hrs = 12 * 60 * 60
+        two_min = 2 * 60
+        node.setmocktime(now + twelve_hrs - two_min)
+        time.sleep(2) # ensure enough time has passed for rebroadcast attempt to occur
         assert_equal(node.p2ps[1].tx_invs_received[txid], 0)
```

Comment:
Would you want to grab the lock? And also since this function is pretty handy?
```suggestion
        assert_equal(txid in node.p2ps[1]get_invs(), False)
```

Reply:
good call, updated.